test/unit/view_model/active_record_test.rb

```diff
@@ -468,4 +468,52 @@ def test_deserialize_context
       assert_nil(ref_error.ref)
     end
   end
+
+    # Parent view should be correctly passed down the tree when deserializing
+  class DeferredConstraintTest < ActiveSupport::TestCase
+    include ARVMTestUtilities
+
+    def before_all
+      super
+
+      build_viewmodel(:List) do
+        define_schema do |t|
+          t.integer :child_id
+        end
+
+        define_model do
+          belongs_to :child, class_name: :List
+        end
+
+        define_viewmodel do
+          association :child, shared: true
+        end
+      end
+      List.connection.execute("ALTER TABLE lists ADD CONSTRAINT unique_child UNIQUE (child_id) DEFERRABLE INITIALLY DEFERRED")
+    end
+
+    class SentinelError < RuntimeError; end
+
+    def test_deferred_constraint_violation
+      l1 = List.create!(child: List.new)
+      l2 = List.create!
+
+      assert_raises(SentinelError) do
```

Comment:
A comment on why this is necessary would be nice. Given that test code is run within a transaction, and the transaction isn't committed, the deferred constraint will only be checked if checked explicitly. It seems like the test is equivalent to 

```ruby
ex = assert_raises { alter_by_view!(...)  {...} }
assert_match(ex, ...)
```